tests/base_tests.rs

```diff
@@ -17,22 +19,37 @@ fn hello_builder() {
     let source_result = xml_parser.parse_file("tests/data/1.xml");
     assert!(source_result.is_ok());
     let source = source_result.unwrap();
+
+    // Parse XSL file from the file system
     let stylesheet_result = xslt_parser::parse_file("tests/data/1.xsl");
     assert!(stylesheet_result.is_ok());
     let mut stylesheet = stylesheet_result.unwrap();
 
+    // Parse XSL file from memory
```

Comment:
you may move these into a new `string_builder` test to have them self-contained, if it's not too much trouble

Reply:
✅ Done, extracted to separate test.

In the future would be nice to also add tests for parameters. There is 1 non-obvious detail there - string parameters must be escaped in single quotes, e.g.:

```rust
    let xslt_params = vec![
        ("yearfrom", "1999"),
        ("yearto", "2000"),
        ("mystringparam", "'value'")
    ];
```

Reply:
In our project we already have a struct and some traits implemented for parameters on the Rust side, which makes it convenient to use with Rust, maybe it makes sense to add them on this library level :)

/// XSLT file accepts parameters that will be applied during transformation.
pub struct XsltParameters {
    params: HashMap<String, String>,
}

```rust
impl XsltParameters {
    pub fn new() -> XsltParameters {
        XsltParameters {
            params: HashMap::new(),
        }
    }

    pub fn get_parameters(&self) -> &HashMap<String, String> {
        &self.params
    }

    /// Insert new parameter or update it if it already exist.
    pub fn set_parameter(&mut self, key: &str, value: &str) -> &mut XsltParameters {
        if let Some(val) = self.params.get_mut(key) {
            *val = value.to_string();
        } else {
            self.params.insert(key.to_string(), value.to_string());
        }

        self
    }

    /// Insert new string parameter or update it if it already exist.
    /// String parameters are handled separately because they should be wrapped in single-quotes, e.g. `"'print'"`.
    pub fn set_string_parameter(&mut self, key: &str, value: &str) -> &mut XsltParameters {
        self.set_parameter(key, self.wrap_with_single_quotes(value).as_str())
    }

    fn wrap_with_single_quotes(&self, value: &str) -> String {
        format!("'{}'", value)
    }
}

impl Default for XsltParameters {
    fn default() -> Self {
        Self::new()
    }
}

impl<'a> From<&'a XsltParameters> for Vec<(&'a str, &'a str)> {
    fn from(params: &'a XsltParameters) -> Self {
        params
            .get_parameters()
            .iter()
            .map(|(k, v)| (k.as_str(), v.as_str()))
            .collect()
    }
}
```

Reply:
Indeed, adding such a convenient abstraction would be useful in my view, and if you already have it - all the better!

The requirement to have the string escapes in the parameters is a little too tricky to my taste. So while your `("mystringparam", "'value'")`  should continue to work, I think a simple `"value"` should also work. That likely means exactly a dedicated `XsltParameters` data structure that does the escaping either when initialized or when serialized for the pass to `xsltApplyStylesheetUser`.

But that can wait for a next PR 👍🏻